tree/ntuple/src/RNTupleDescriptor.cxx

```diff
@@ -169,6 +169,23 @@ bool ROOT::RFieldDescriptor::IsCustomClass() const
    return true;
 }
 
+bool ROOT::RFieldDescriptor::IsCustomEnum(const RNTupleDescriptor &desc) const
+{
+   if (fStructure != ROOT::ENTupleStructure::kPlain)
+      return false;
+   if (fTypeName.rfind("std::", 0) == 0)
+      return false;
+
+   auto subFieldId = desc.FindFieldId("_0", fFieldId);
+   if (subFieldId == kInvalidDescriptorId)
+      return false;
+
+   static const std::unordered_set<std::string> gIntTypeNames = {"char",          "std::int8_t",   "std::uint8_t",
```

Comment:
Can't `bool` also be a valid option here?

Reply:
Are you sure? It seems supported by both [clang](https://godbolt.org/z/vhe3WherG) and [GCC](https://godbolt.org/z/evaaPEneq)...

Reply:
Yep, enum works with integer types, and bool is an (extended) integer type (0 and 1), much like char.
Related: https://stackoverflow.com/a/68321667/7471760

Reply:
See #19924

Reply:
and fixed here, too.